torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1884,6 +1884,41 @@ def _lazy_init(self) -> None:
         # to non-root instances
         inconsistent_limit_all_gathers = False
         for fsdp_module in self.fsdp_modules(self):
+            if not fsdp_module._use_orig_params and fsdp_module._has_params:
+                # Check if the wrapped module changed after construction
+                # (e.g. applying the activation checkpointing wrapper) and
+                # if so, de-register the `FlatParameter` from the old
+                # wrapped module and register it to the new wrapped module
+                flat_param = fsdp_module._handles[0].flat_param
+                target_submodule = None
+                target_name = None
+                for submodule in fsdp_module.modules():
+                    for param_name, param in submodule._parameters.items():
+                        if flat_param is param:
+                            target_submodule = submodule
+                            target_name = param_name
+                            break
```

Comment:
This only break the inner loop. Although there is only target_submodule, we should also break the outer loop to make the code less confusion.